src/protocols/user_directed_commissioning/UserDirectedCommissioningServer.cpp

```diff
@@ -0,0 +1,145 @@
+{
+    msgBuf->DebugDump("NEW UserDirectedCommissioningServer::OnMessageReceive");
+    ChipLogProgress(AppServer, "NEW UserDirectedCommissioningServer::OnMessageReceived");
+
+    PacketHeader packetHeader;
+
+    ReturnOnFailure(packetHeader.DecodeAndConsume(msgBuf));
+
+    if (packetHeader.GetFlags().Has(Header::FlagValues::kEncryptedMessage))
+    {
+        ChipLogError(AppServer, "UDC encryption flag set - ignoring");
+        return;
+    }
+
+    // TODO: do we need these std::move calls?
+    System::PacketBufferHandle && msg = std::move(msgBuf);
+
+    PayloadHeader payloadHeader;
+    ReturnOnFailure(payloadHeader.DecodeAndConsume(msg));
+
+    System::PacketBufferHandle && payload = std::move(msg);
+
+    char instanceName[chip::Mdns::kMaxInstanceNameSize + 1];
+    size_t instanceNameLength =
+        (payload->DataLength() > (chip::Mdns::kMaxInstanceNameSize)) ? chip::Mdns::kMaxInstanceNameSize : payload->DataLength();
+    payload->Read((uint8_t *) instanceName, instanceNameLength);
+
+    instanceName[instanceNameLength] = '\0';
+
+    ChipLogProgress(AppServer, "UDC instance=%s", instanceName);
+
+    UDCClientState * client = mUdcClients.FindUDCClientState(instanceName, nullptr);
+    if (client == nullptr)
+    {
+        ChipLogProgress(AppServer, "UDC new instance state received");
+
+        CHIP_ERROR err;
+        err = mUdcClients.CreateNewUDCClientState(instanceName, &client);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "UDC error creating new connection state");
+            return;
+        }
+        if (client == nullptr)
+        {
+            ChipLogError(AppServer, "UDC no memory");
+            return;
+        }
+
+        // Call the registered InstanceNameResolver, if any.
+        if (mInstanceNameResolver != nullptr)
+        {
+            mInstanceNameResolver->FindCommissionableNode(instanceName);
+        }
+        else
+        {
+            ChipLogError(AppServer, "UserDirectedCommissioningServer::OnMessageReceived no mInstanceNameResolver registered");
+        }
+    }
+
+    mUdcClients.MarkUDCClientActive(client);
+}
+
+void UserDirectedCommissioningServer::SetUDCClientProcessingState(char * instanceName, UDCClientProcessingState state)
+{
+    UDCClientState * client = mUdcClients.FindUDCClientState(instanceName, nullptr);
+    if (client == nullptr)
+    {
+        // printf("SetUDCClientProcessingState new instance state received\n");
+        CHIP_ERROR err;
+        err = mUdcClients.CreateNewUDCClientState(instanceName, &client);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer,
+                         "UserDirectedCommissioningServer::SetUDCClientProcessingState error creating new connection state");
+            return;
+        }
+        if (client == nullptr)
+        {
+            ChipLogError(AppServer, "UserDirectedCommissioningServer::SetUDCClientProcessingState no memory");
```

Comment:
Again, wouldn't that have returned error?

Reply:
addressed in latest changes